celltype_annotation/analysis/cell-assign-sarcoma-marker-genes.Rmd

```diff
@@ -0,0 +1,194 @@
+---
+title: "CellAssign: Rhabdomyosarcoma using marker genes"
+output: 
+  html_notebook:
+    toc: true
+    toc_float: true
+params:
+  library_id: "SCPCL000488"
+  sample_id: "SCPCS000262"
+---
+
+Since it's fairly obvious that `CellAssign`, although better at assigning `NA` than `SingleR`, is still going to be highly dependent on the marker genes that are in the reference, I wanted to see how well `CellAssign` performs when we provide marker genes that we know are associated with the cell types found in the dataset.
+
+Here I use `scran::FindMarkers()` to get the top markers for each of the cell types and then use that as the reference gene by cell matrix for `CellAssign`.
+I expect that the output of `CellAssign` should be fairly consistent with the annotations we have using this method.
+
+## Set Up
+
+```{r}
+library(SingleCellExperiment)
+library(ggplot2)
+```
+
+```{r}
+# Set up paths 
+
+# build path to annotated sce file 
+processed_sce_file <- glue::glue("{params$library_id}_processed_celltype.rds")
+processed_sce_path <- file.path("..", "data", params$sample_id, processed_sce_file)
+
+# output matrix files 
+rms_matrix_file <- file.path("..", "references", "rms_broad_markers_mtx.csv")
+
+# anndata input
+anndata_dir <- file.path("..", "data", "anndata")
+anndata_file <- glue::glue("{params$library_id}_anndata.hdf5")
+anndata_file <- file.path(anndata_dir, anndata_file)
+fs::dir_create(anndata_dir)
+
+# cellassign predictions
+rms_prediction_file <- file.path(
+  anndata_dir, 
+  glue::glue("{params$library_id}_rms_broad_markers.tsv")
+)
+
+```
+
+```{r}
+# read in all annotated sce 
+processed_sce <- readr::read_rds(local_processed_sce_path)
+```
+
+```{r}
+# source helper functions
+function_file <- file.path("..", "utils", "cellassign-helper-functions.R")
+source(function_file)
+```
+
+## Get marker genes
+
+```{r}
+# extract coldata for easy plotting
+coldata_df <- colData(processed_sce) |>
+  as.data.frame() |> 
+  dplyr::select(barcode, celltype) |>
+  # classify cells as tumor or normal
+  dplyr::mutate(cell_category = dplyr::if_else(stringr::str_detect(celltype, "Tumor"), "Tumor", celltype),
+                # remove sub states of myoblast/mesoderm/myocyte
+                broad_celltype = stringr::str_remove(celltype, "-A|-B|-C|-D"))
+```
+
+
+```{r}
+# find markers for all cell types 
```

Comment:
Worth noting that you are looking for markers that are "unique" to each cell type. (This may be a reason for some uncertainty: there is no ability to do any kind of "hierarchical" model. 

(This has me doing some deeper thinking about how best to select markers automatically, but I don't have any answers, and this is not where to explore it!)